### PR TITLE
MD-360: implement cancelling query after timeout

### PR DESCRIPTION
**Pull request by cchang738**
@sudheeshkatkam 

Can you review the PR as we discussed? I checked, after statement.cancel(), no other call can be made to check status.
Thanks.
http://10.10.104.91:8080/job/Framework-Regression/508/console

### CODE REVIEW

File: framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java [hunk 1]
```diff
@@ -300,6 +301,13 @@ private boolean isNullabilityCompatible(List<Integer> limitZero, List<Integer> r
...
   @Override
   public void cancel() {
+	if (statement != null)
+		try {
+			statement.cancel();
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
     thread.interrupt();
```
Comment (sudheeshkatkam) [inline framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java]: If the query was cancelled successfully, the `thread` will be unblocked in near future. So this `interrupt()` call could "hide" incorrect handling of cancellation.

File: framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java [hunk 2]
```diff
@@ -226,6 +228,7 @@ private void executeQuery(String query) throws IOException, SQLException {
           writer.write(columnList + "\n");
         }
       }
+      doneProcessingResultSet = true;
```
Comment (sudheeshkatkam) [inline framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java]: Use an AtomicBoolean, and move this statement to finally block.

File: framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java [hunk 3]
```diff
@@ -300,8 +303,30 @@ private boolean isNullabilityCompatible(List<Integer> limitZero, List<Integer> r
...
+		e.printStackTrace();
+		return;
+	  }
+	  i++;
+	}
+	if (!doneProcessingResultSet) {
+      
+	  LOG.warn("Cancel after timeout may have failured!");
+	}
+	thread.interrupt();
```
Comment (sudheeshkatkam) [inline framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java]: Statement should be in the if condition block.

**Comment by sudheeshkatkam** [comment]
Comments to be addressed. Otherwise, +1

File: framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java [hunk 4]
```diff
@@ -300,8 +303,30 @@ private boolean isNullabilityCompatible(List<Integer> limitZero, List<Integer> r
...
+		thread.sleep(1000);
+	  } catch (InterruptedException e) {
+		e.printStackTrace();
+		return;
+	  }
+	  i++;
+	}
+	if (!doneProcessingResultSet) {
+      
+	  LOG.warn("Cancel after timeout may have failured!");
```
Comment (sudheeshkatkam) [inline framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java]: failed*

File: framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java [hunk 5]
```diff
@@ -300,8 +305,30 @@ private boolean isNullabilityCompatible(List<Integer> limitZero, List<Integer> r
...
+		statement.cancel();
+	  } catch (SQLException e) {
+		LOG.warn("Cancel after timeout failed!");
+		e.printStackTrace();
+	  }
+	}
+	int i = 0;
+	while (!doneProcessingResultSet.get() && i < 10) {
+	  try {
+		thread.sleep(1000);
```
Comment (sudheeshkatkam) [inline framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java]: Current thread should sleep, not `thread`.

Comment (cchang738) [inline framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java]: Good catch. Thanks.

**Pull request merged**